### PR TITLE
Ac 2459 add signature type

### PR DESCRIPTION
**Pull request by soeren-schmaljohann-2denker**

### CODE REVIEW

File: packages/apptive_grid_core/lib/src/model/data_entity.dart [hunk 1]
```diff
@@ -520,3 +522,18 @@ class PhoneNumberDataEntity extends DataEntity<String, String> {
...
+  SignatureDataEntity([super.value]);
+
+  /// Creates a new SignatureDataEntity from a Json Response
+  factory SignatureDataEntity.fromJson(dynamic jsonValue) =>
+      SignatureDataEntity(
+        jsonValue != null ? Attachment.fromJson(jsonValue) : null,
+      );
+
+  @override
+  dynamic get schemaValue => value != null ? value!.toJson() : null;
```
Comment (ABausG) [inline packages/apptive_grid_core/lib/src/model/data_entity.dart:538]: Could just be `value?.toJson()`

File: packages/apptive_grid_form/lib/src/widgets/form_widget/signature_form_widget.dart [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+      _isLoadingPreviousValue = false;
+    }
+
+    _signatureController.addListener(() => setState(() {}));
+  }
+
+  @override
+  void dispose() {
+    super.dispose();
+    _signatureController.dispose();
```
Comment (ABausG) [inline packages/apptive_grid_form/lib/src/widgets/form_widget/signature_form_widget.dart:51]: Other way around

File: packages/apptive_grid_form/lib/src/widgets/form_widget/signature_form_widget.dart [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+      },
+    ).then(
+      (didConfirm) {
+        setSignature(didConfirm == true ? tmpSignatureControl.paths : null);
+        tmpSignatureControl.dispose();
+      },
+    );
+  }
+
+  void setSignature(List<CubicPath>? paths) async {
```
Comment (ABausG) [inline packages/apptive_grid_form/lib/src/widgets/form_widget/signature_form_widget.dart:160]: Might be private

File: packages/apptive_grid_core/lib/src/model/data_entity.dart (hunk 1 above)
Comment (soeren-schmaljohann-2denker) [reply packages/apptive_grid_core/lib/src/model/data_entity.dart]: done

File: packages/apptive_grid_form/lib/src/widgets/form_widget/signature_form_widget.dart (hunk 2 above)
Comment (soeren-schmaljohann-2denker) [reply packages/apptive_grid_form/lib/src/widgets/form_widget/signature_form_widget.dart]: done

File: packages/apptive_grid_form/lib/src/widgets/form_widget/signature_form_widget.dart (hunk 3 above)
Comment (soeren-schmaljohann-2denker) [reply packages/apptive_grid_form/lib/src/widgets/form_widget/signature_form_widget.dart]: done

**Review by ABausG — APPROVED** [review]

**Pull request merged**